core/src/kura.rs

```diff
@@ -340,8 +379,8 @@ pub trait BlockStoreTrait {
     ///
     /// # Errors
     /// IO Error.
-    fn read_block_index(&self, block_height: u64) -> Result<(u64, u64)> {
-        let mut index = (0, 0);
+    fn read_block_index(&self, block_height: u64) -> Result<(u64, u64, u64)> {
+        let mut index = (0, 0, 0);
```

Comment:
you could be a smartass here and use the first bit of the block start to store the information about the format type. That is unless we'll support more serialization formats that we can change mid stream while blockchain is up

Reply:
I considered that. But I wanted to preserve this ability, if you only ever store plain text blocks the `blocks.data` file is a continuous text file without any funny characters or invalid unicode. There is space in the block index now for additional formats and information. All 63 bits per block.